dev/reference/alert-rules.md

```diff
@@ -0,0 +1,1276 @@
+
+#### `TiKV_space_used_more_than_80%`
+
+* Alert rule:
+
+    `sum(pd_cluster_status{type="storage_size"}) / sum(pd_cluster_status{type="storage_capacity"}) * 100 > 80`
+
+* Description:
+
+    Over 80% of the cluster space is occupied.
+
+* Solution:
+
+    * Check whether it is needed to increase capacity.
+    * Check whether there is any file that occupies a large amount of disk space, such as the log, snapshot, core dump, etc.
+
+#### `PD_system_time_slow`
+
+* Alert rule:
+
+    `changes(pd_server_tso{type="system_time_slow"}[10m]) >= 1`
+
+* Description:
+
+    The system time rewind might happen.
+
+* Solution:
+
+    Check whether the system time is configured correctly.
+
+#### `PD_no_store_for_making_replica`
+
+* Alert rule:
+
+    `increase(pd_checker_event_count{type="replica_checker", name="no_target_store"}[1m]) > 0`
+
+* Description:
+
+    There is no appropriate store for additional replicas.
+
+* Solution:
+
+    * Check whether there is enough space in the store.
+    * Check whether there is any store for additional replicas according to the label configuration if it is configured.
+
+## TiKV alert rules
+
+This section gives the alert rules for the TiKV component.
+
+### Emergency-level alerts
+
+Emergency-level alerts are often caused by a service or node failure. Manual intervention is required immediately.
+
+#### `TiKV_memory_used_too_fast`
+
+* Alert rule:
+
+    `process_resident_memory_bytes{job=~"tikv",instance=~".*"} - (process_resident_memory_bytes{job=~"tikv",instance=~".*"} offset 5m) > 5*1024*1024*1024`
+
+* Description:
+
+    Currently, there are no TiKV monitoring items that are about memory. You can monitor the memory usage of the machines in the cluster by node_exporter. The above rule indicates that when the memory usage exceeds 5 GB within 5 minutes (the memory is occupied too fast in TiKV), an alert is triggered.
+
+* Solution:
+
+    Adjust the `block-cache-size` of `rockdb.defaultcf` and `rocksdb.writecf`.
+
+#### `TiKV_GC_can_not_work`
+
+* Alert rule:
+
+    `sum(increase(tidb_tikvclient_gc_action_result{type="success"}[6h])) < 1`
+
+* Description:
+
+    GC is not performed successfully in a Region within 6 hours, which indicates that GC is not working properly. If GC does not run in a short term, it will not cause much trouble; but if GC keeps down, more and more versions are retained, which slows down the query.
+
+* Solution:
+
+    1. Perform `select VARIABLE_VALUE from mysql.tidb where VARIABLE_NAME = "tikv_gc_leader_desc"` to locate the `tidb-server` corresponding to the GC leader;
+    2. View the log of the `tidb-server`, and grep gc_worker tidb.log;
+    3. If you find that the GC worker has been resolving locks (the last log is "start resolve locks") or deleting ranges (the last log is “start delete {number} ranges”) during this time, it means the GC process is running normally. Otherwise, contact `support@pingcap.com` to resolve this issue.
```

Comment:
```suggestion
    3. If you find that the GC worker has been resolving locks (the last log is "start resolve locks") or deleting ranges (the last log is “start delete {number} ranges”) during this time, it means the GC process is running normally. Otherwise, contact [support@pingcap.com](mailto:support@pingcap.com) to resolve this issue.
```